src/client/Microsoft.Identity.Client/AppConfig/CiamAuthorityHelper.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Internal;
+
+namespace Microsoft.Identity.Client.AppConfig
+{
+    /// <summary>
+    /// Translates CIAM authorities into a usable form. This is needed only until ESTS is updated to support the north star format
```

Comment:
Since the users can still pass in the transformed format, would we have to reverse this code then to transform into North Star format or in the future will AAD support both formats?

Reply:
Not sure about that yet. That will depend on the decisions made by management. @jmprieur any thoughts on this?
I would imagine ESTS would allow the transformed versions ('https://tenant.ciamlogin.com/tenant.onmicrosoft.com`) to continue working going forward as to not break the customers who are using the version of MSAL at build.